polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -569,6 +554,29 @@ impl<T: Config> Pallet<T> {
 
 		METRICS.on_candidates_processed_total(backed_candidates.len() as u64);
 
+		// After freeing cores and filling claims, but before processing backed candidates
```

Comment:
why do this after we update the claim queue?
isn't this our parent's claim queue?

Reply:
oh but our parent did not advance its claim queue at the end of the block. so yeah this looks right

Reply:
thinking about this again.
the claim queue of our parent cannot take into account the bitfields of the child. (we only need to have claims filled)

So this claim queue will be different than what the claim queue runtime API had returned at our parent.
We should really update the tracked claim queue considering no bitfields (as we do in the claim queue runtime API)

btw, this is fixed in https://github.com/paritytech/polkadot-sdk/pull/5461 by filling the claim queue at the end of the block. which also enables us to remove the queue filling call in the runtime APIs

Reply:
@sandreim (tagging so that the comment does not get lost)

Reply:
oh nevermind, bitfields don't influence the claim queue here. we pop from it later based on the freed cores.